server/webhook_test.go

```diff
@@ -0,0 +1,88 @@
+package server
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSendToWebhookIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	s := &Server{
+		StartTime: time.Now(),
+	}
+
+	validPayload := &Payload{Username: "mattermod", Text: "test"}
+	expectedMattermostPayload := "ok"
+	mattermost := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte(expectedMattermostPayload))
+		if err != nil {
+			log.Fatal(err)
+		}
```

Comment:
I wanted to validate the incoming payload. Not the outgoing response. :)

Validating that we are returning `ok` is fine but is not very important. It's a response from an external server. But our objective is to validate our code. i.e. we want to verify that we are sending the correct data to the server. Let's do that.

Reply:
I was actually confused. I didn't want to validate that payload.